apps/minifront/src/components/header/top-row.tsx

```diff
@@ -13,20 +13,18 @@ import { PagePath } from '../metadata/paths';
 import { TabletNavMenu } from './tablet-nav-menu';
 import { useEffect, useState } from 'react';
 import { getChainId } from '../../fetchers/chain-id';
-import { LayoutLoaderResult } from '../layout';
 
 // Infinite-expiry invite link to the #web-ext-feedback channel. Provided by
 // Henry (@hdevalence) and thus tied to his Discord account, so reach out to him
 // if there are any problems with this link.
 const WEB_EXT_FEEDBACK_DISCORD_CHANNEL = 'https://discord.gg/XDNcrhKVwV';
 
 export const TopRow = () => {
-  const { isInstalled, isConnected } = useLoaderData() as LayoutLoaderResult;
   const [chainId, setChainId] = useState<string | undefined>();
 
   useEffect(() => {
-    if (isInstalled && isConnected) void getChainId().then(setChainId);
```

Comment:
Is there a reason why we don't need this guard anymore? Is it now impossible to get this component unless already connected?

Reply:
this header doesn't appear on the connection error pages, which should be the only thing that appears if it's not possible to connect to the extension.